accounts-cluster-bench/src/main.rs

```diff
@@ -227,6 +229,264 @@ fn make_close_message(
+        let chunk_end = std::cmp::min(chunk_start + chunk_size, end);
+
+        let addresses: Vec<_> = (chunk_start..chunk_end)
+            .map(|seed| {
+                Pubkey::create_with_seed(base_keypair_pubkey, &seed.to_string(), program_id)
+                    .unwrap()
+            })
+            .collect();
+        chunk_start = chunk_end;
+        let mut rpc_time = Measure::start("rpc-get-multiple-accounts");
+        match client.get_multiple_accounts(&addresses) {
+            Ok(accounts) => {
+                rpc_time.stop();
+                for account in accounts.into_iter().flatten() {
+                    if thread_rng().gen_ratio(1, 10_000) {
+                        info!(
+                            "account: lamports {:?} size: {} owner: {:?}",
+                            account.lamports,
+                            account.data.len(),
+                            account.owner
+                        );
+                    }
+                }
+                stats.total_success_time_us += rpc_time.as_us();
+                stats.success += 1;
+            }
+            Err(e) => {
+                rpc_time.stop();
+                stats.total_errors_time_us += rpc_time.as_us();
+                stats.errors += 1;
+                if last_error.elapsed().as_secs() > 1 {
+                    info!("error: {:?}", e);
+                    *last_error = Instant::now();
+                }
+                debug!("error: {:?}", e);
+            }
+        }
+    }
+}
+
+#[derive(Default)]
+struct RpcBenchStats {
+    errors: u64,
+    success: u64,
+    total_errors_time_us: u64,
+    total_success_time_us: u64,
+}
+
+fn run_rpc_bench_loop(
+    rpc_bench: RpcBench,
+    thread: usize,
+    client: &RpcClient,
+    base_keypair_pubkey: &Pubkey,
+    exit: &AtomicBool,
+    program_id: &Pubkey,
+    max_closed: &AtomicU64,
+    max_created: &AtomicU64,
+    mint: &Option<Pubkey>,
+) {
+    let mut stats = RpcBenchStats::default();
+    let mut iters = 0;
+    let mut last_error = Instant::now();
+    let mut last_print = Instant::now();
+    loop {
+        if exit.load(Ordering::Relaxed) {
+            break;
+        }
+        match rpc_bench {
+            RpcBench::GetSlot => {
+                let mut rpc_time = Measure::start("rpc-get-slot");
+                match client.get_slot() {
+                    Ok(_slot) => {
+                        rpc_time.stop();
+                        stats.success += 1;
+                        stats.total_success_time_us += rpc_time.as_us();
+                    }
+                    Err(e) => {
+                        rpc_time.stop();
+                        stats.total_errors_time_us += rpc_time.as_us();
+                        stats.errors += 1;
+                        if last_error.elapsed().as_secs() > 2 {
+                            info!("get_slot error: {:?}", e);
+                            last_error = Instant::now();
+                        }
```

Comment:
Maybe dedupe this into a closure or fn in case we want to twiddle the frequency or make other changes to all RpcBench variants?

Reply:
I can do this in a follow-up, requires some re-shuffling.